app/app.go

```diff
@@ -755,7 +757,16 @@ func (app *App) RegisterAPIRoutes(apiSvr *api.Server, _ config.APIConfig) {
 func (app *App) RegisterTxService(clientCtx client.Context) {
 	authtx.RegisterTxService(app.BaseApp.GRPCQueryRouter(), clientCtx, app.BaseApp.Simulate, app.interfaceRegistry)
 	celestiatx.RegisterTxService(app.BaseApp.GRPCQueryRouter(), clientCtx, app.interfaceRegistry)
-	gasestimation.RegisterGasEstimationService(app.BaseApp.GRPCQueryRouter(), clientCtx, app.BaseApp.Simulate)
+	gasestimation.RegisterGasEstimationService(app.BaseApp.GRPCQueryRouter(), clientCtx, app.txConfig.TxDecoder(), app.getGovMaxSquareBytes, app.BaseApp.Simulate)
+}
+
+func (app *App) getGovMaxSquareBytes() (uint64, error) {
+	ctx, err := app.CreateQueryContext(app.LastBlockHeight(), false)
+	if err != nil {
+		return 0, err
+	}
+	maxSquareSize := app.BlobKeeper.GovMaxSquareSize(ctx)
+	return maxSquareSize * maxSquareSize * share.ContinuationSparseShareContentSize, nil
```

Comment:
[nit] this still uses a constant (`share.ContinuationSparseShareContentSize`) that is an implementation detail. Given the estimate and logic around this code is not exact, IMO it would be clearer to use `share.ShareSize`.

Reply:
aef21be44da31797e2fcc878ecc3e804290a9429